### PR TITLE
Add complex sample

### PR DESCRIPTION
**Pull request by benkoshy**
I thought I'd add in a complex query based on: https://github.com/jeremyevans/sequel/discussions/1939

I have found that reading examples, especially complex ones with joins etc. is particularly helpful. 

Due care has been taken, but it is possible i've made a mistake in the sql output.

### CODE REVIEW

File: doc/querying.rdoc [hunk 1]
```diff
@@ -715,6 +715,20 @@ aggregation:
...
   # SELECT artist_id, sum(num_tracks) AS tracks FROM albums GROUP BY artist_id
+  
+You can combine select_group, select_append while we are using a join and a where statement:
+
+  Album.
+  left_join(:artists.
+      select_group(:id).
+      select_append{max(:sales)},
+    :id=>:album_id).
+  where{{Album[:song] => "Bohemian Rhapsody"}}
```
Comment (jeremyevans) [inline doc/querying.rdoc:726]: You probably want `albums[:song]`, not `Album[:song]`, as evidenced by the SQL produced.

File: doc/querying.rdoc [hunk 2]
```diff
@@ -715,6 +715,20 @@ aggregation:
...
+  
+You can combine select_group, select_append while we are using a join and a where statement:
+
+  Album.
+  left_join(:artists.
+      select_group(:id).
+      select_append{max(:sales)},
+    :id=>:album_id).
+  where{{Album[:song] => "Bohemian Rhapsody"}}  
+  # SELECT * FROM `albums`
```
Comment (jeremyevans) [inline doc/querying.rdoc:727]: Let's drop the MySQL/SQLite specific quoting.  Probably best to show an unquoted example.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
I'm OK with the general idea.  Please see inline comments.

File: doc/querying.rdoc (hunk 2 above)
Comment (benkoshy) [reply doc/querying.rdoc]: I've removed all back-ticks. Pls LMK if this is wrong.

File: doc/querying.rdoc [hunk 3]
```diff
@@ -715,6 +715,20 @@ aggregation:
 
   Album.select_group(:artist_id).select_append{sum(num_tracks).as(tracks)}
   # SELECT artist_id, sum(num_tracks) AS tracks FROM albums GROUP BY artist_id
+  
+You can combine select_group, select_append while we are using a join and a where statement:
+
+  Album.
+  left_join(:artists.
```
Comment (jeremyevans) [inline doc/querying.rdoc:722]: Since you are left joining to a subselect, I'm guessing this should be `DB[:artists]` or `Artist`.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Found a couple more issues.  It's best to double check your work using a mock database (`bin/sequel` with no arguments gives you a mock database in `DB`):

```
bin/sequel -c 'puts DB[:albums].
  left_join(DB[:artists].
      select_group(:id).
      select_append{max(:sales)},
    :id=>:album_id).
  where{{albums[:song] => "Bohemian Rhapsody"}}.sql'

SELECT * FROM albums
LEFT JOIN (SELECT id, max(sales) FROM artists GROUP BY id) AS t1
  ON (t1.id = albums.album_id)
WHERE (albums.song = 'Bohemian Rhapsody')
```

File: doc/querying.rdoc [hunk 4]
```diff
@@ -715,6 +715,20 @@ aggregation:
...
+      select_group(:id).
+      select_append{max(:sales)},
+    :id=>:album_id).
```
[the remark this answers is not in the document]
Comment (benkoshy) [reply doc/querying.rdoc:725]: switched the order.

**Comment by benkoshy** [comment]
Fixed using `bin/sequel` runner which was inordinately helpful in producing the output. Apologies for the rigmarole - I thank you for your patience. 

Not sure why CI is failing though.

**Comment by jeremyevans** [comment]
Look good, I'll merge soon.  The CI failures look to be based on a bundle issue trying to install a newer version of rake than supported on old Ruby versions.  I'll fix that.

**Pull request merged**